src/group_theory/group_action/sub_mul_action.lean

```diff
@@ -108,21 +115,36 @@ end has_scalar
 
 section mul_action
 
-variables [monoid R]
+variables [monoid S] [monoid R]
 
 variables [mul_action R M]
+variables [has_scalar S R] [mul_action S M] [is_scalar_tower S R M]
 variables (p : sub_mul_action R M)
 variables {r : R} {x : M}
 
-@[simp] lemma smul_mem_iff' (u : units R) : (u : R) • x ∈ p ↔ x ∈ p :=
-⟨λ h, by simpa only [smul_smul, u.inv_mul, one_smul] using p.smul_mem ↑u⁻¹ h, p.smul_mem u⟩
+lemma smul_of_tower_mem (s : S) (h : x ∈ p) : s • x ∈ p :=
+by { rw [←one_smul R x, ←smul_assoc], exact p.smul_mem _ h }
```

Comment:
What do you think about making `smul_mem` itself assume `is_scalar_tower`? So the `smul_mem'` field in `sub_mul_action` is still about `smul` with `R`, and then `smul_mem` is about `smul` with `S`.

Reply:
That's what I tried already in #5569. It's quite an invasive change though, so I decided that it was better to do it in a separate PR.